### ISSUE TITLE
Adding a custom rule

### ISSUE THREAD

**Issue by QADEvops1** [mwz/sonar-scala#49]
Hi,

I have added a custom rule.  Please see below.

![image](https://user-images.githubusercontent.com/36966485/38639256-9d520680-3d9e-11e8-9586-19ac6e454b05.png)

When I do the scan this rule is not working (I have hardcoded IP address in my code).  Can you please help.

**Comment by BalmungSan** [comment]
Hello @QADEvops1, I think you need to activate the rule in the profile you're using to scan the project.
(Well I'm assuming that "not working" means that the rule was no used during the scan, if that is not the problem, could you provide an explanation of what went wrong)

**Comment by QADEvops1** [comment]
Hi,

The rule is activated.  Yes the rule was not used during scan.  It shows in
list of rules available in quality profile.

**Comment by QADEvops1** [comment]
Yes. Please see the screenshot in the 1st comment.

**Comment by BalmungSan** [comment]
Just to validate that the problem is not in the regex, could you create another custom rule that checks for a simple pattern ?, something like `class` or `String`. So that the rule should be reported massively ?

If the problem persist I will try to reproduce and debug it.

**Comment by BalmungSan** [comment]
Hello @QADEvops1, as mentioned in the previous comment, I worked for me.
Following are some screenshots showing it.

![image](https://user-images.githubusercontent.com/11336950/38646869-5c844bb6-3daf-11e8-9c08-c45801fe67f4.png)

![image](https://user-images.githubusercontent.com/11336950/38646950-a86805a4-3daf-11e8-85e5-462968ca6cd6.png)

So, now I'm wondering, why it didn't worked for you. 🤔 
Could you provide something to reproduce the problem, a sample file or something ?

**Comment by Trif3cta** [comment]
Hi BalmungSan -- thanks for your help so far.  I'm working with user QADevops1 to get this working and glad to see it can work.  Here is an excerpt from a file we're attempting to scan; as you'll see, this part is nonsensical but should be detected nevertheless:

Our installed plug-in version is 6.0.0.

```
  /** test code */
  def TCP = 192.168.1.0
  def abc = 192.168.1:9000
  def localhost = "http://localhost"
  def LocalHost
  def URL = "tcp://10.9.21.202:17222"
```

**Comment by BalmungSan** [comment]
Hello @Trif3cta, I'll take a look in a couple of hours.

Just a few questions:

+ The second `def LocalHost` is empty, that was a typing mistake ?, because there you have a **syntax error**

+ Which ones are being detected and which ones don't ?, or any of them are working ?

**Comment by Trif3cta** [comment]
Hi -- correct, the file will not compile because those "keywords" just typed into the file like TCP and localhost were created for other rules just looking for that text, which is separate from the IP rule -- so none of the 3 rules work.  Maybe the file needs to be compile-able?

**Comment by BalmungSan** [comment]
Hello @Trif3cta & @QADEvops1.
I want to apologize for the delay, but yesterday was a difficult day...

Anyways, I had just checked with your examples, and, as can see below, effectively they didn't work.
![image](https://user-images.githubusercontent.com/11336950/38737398-b0a384c8-3ef4-11e8-880f-0e9a4b70c46d.png)

Whereas the previous example is still working.
![image](https://user-images.githubusercontent.com/11336950/38737441-d1fa25c8-3ef4-11e8-86bf-e4422c86ae7f.png)

Additionally I got the following output when running the **sonar-scanner** command with the **-X** flag _(debug mode)_.

```bash
13:23:23.770 INFO: Project key: test
13:23:23.771 DEBUG: Start recursive analysis of project modules
13:23:23.771 INFO: -------------  Scan Test
13:23:23.930 INFO: Load server rules
13:23:24.052 DEBUG: GET 200 http://ec2-13-59-211-71.us-east-2.compute.amazonaws.com/anthill/sonarqube/api/rules/list.protobuf | time=122ms
13:23:24.145 INFO: Load server rules (done) | time=215ms
13:23:24.206 INFO: Base dir: /home/balmungsan/test
13:23:24.206 INFO: Working dir: /home/balmungsan/test/.scannerwork
13:23:24.208 INFO: Source paths: src/main/scala
13:23:24.208 INFO: Source encoding: UTF-8, default locale: en_US
13:23:24.255 DEBUG: Declared extensions of language Scala were converted to sonar.lang.patterns.scala : **/*.scala
13:23:24.258 DEBUG: Initializers : 
13:23:24.269 INFO: Index files
13:23:24.311 DEBUG: 'src/main/scala/test/Main.scala' indexed with language 'scala'
13:23:24.312 INFO: 1 file indexed
13:23:24.312 INFO: Quality profile for scala: Test
13:23:24.376 DEBUG: 'Generic Coverage Report' skipped because one of the required properties is missing
13:23:24.377 DEBUG: 'Generic Test Executions Report' skipped because one of the required properties is missing
13:23:24.394 DEBUG: 'Generic Coverage Report' skipped because one of the required properties is missing
13:23:24.394 DEBUG: 'Generic Test Executions Report' skipped because one of the required properties is missing
13:23:24.399 DEBUG: Sensors : com.ncredinburgh.sonar.scalastyle.ScalastyleSensor@1ddd3478 -> Scala Sensor -> Scoverage Sensor -> Zero Coverage Sensor -> CPD Block Indexer
13:23:24.400 INFO: Sensor com.ncredinburgh.sonar.scalastyle.ScalastyleSensor [scala]
13:23:24.506 DEBUG: Using config ScalastyleConfiguration(sonar,true,List(ConfigurationChecker(org.scalastyle.file.RegexChecker,ErrorLevel,true,Map(scalastyle-checker -> org.scalastyle.file.RegexChecker, regex -> (?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?), line -> true),None,Some(IP_Checker_2))))
13:23:24.901 ERROR: Got exception message from Scalastyle. Check you have valid parameters configured for all rules. Exception message was: Expected token SEMI but got Token(FLOATING_POINT_LITERAL,.1,78,.1)
13:23:24.901 INFO: Sensor com.ncredinburgh.sonar.scalastyle.ScalastyleSensor [scala] (done) | time=501ms
13:23:24.901 INFO: Sensor Scala Sensor [scala]
13:23:24.954 INFO: Sensor Scala Sensor [scala] (done) | time=53ms
```

> **13:23:24.901 ERROR: Got exception message from Scalastyle. Check you have valid parameters configured for all rules. Exception message was: Expected token SEMI but got Token(FLOATING_POINT_LITERAL,.1,78,.1)**

So, now I'm pretty sure that the rule is failing because the file is not compile-able, due syntax errors.

**Comment by Trif3cta** [comment]
Hi -- no worries, this was very helpful (also the -X), and it's likely to carry over into other language rule scanners -- that is, the code needs to compile without error.  We'll try on a different set of code and if there are any problems, we'll report back.  Thanks again.
